File: StudentService.java
package com.example.Register;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class StudentService {

    private final StudentRepository studentRepository;

    @Autowired
    public StudentService(StudentRepository studentRepository) {
        this.studentRepository = studentRepository;
    }

    // Method to retrieve all students
    public List<StudentEntity> getAllStudents() {
        return studentRepository.findAll();
    }

    // Method to retrieve a student by ID
    public Optional<StudentEntity> getStudentById(Long id) {
        return studentRepository.findById(id);
    }

    // Method to create a new student
    public StudentEntity createStudent(StudentEntity student) {
        // Ensure the ID is null to prevent overwriting existing records
        return studentRepository.save(student);
    }

    // Method to update an existing student
    public Optional<StudentEntity> updateStudent(Long id, StudentEntity updatedStudent) {
        return studentRepository.findById(id).map(existingStudent -> {
            // Update the existing student with the new data
            existingStudent.setName(updatedStudent.getName());
            existingStudent.setAddress(updatedStudent.getAddress());
            existingStudent.setParentName(updatedStudent.getParentName());
            existingStudent.setPhoneNumber(updatedStudent.getPhoneNumber());
            existingStudent.setStudentClass(updatedStudent.getStudentClass());
            return studentRepository.save(existingStudent);
        });
    }

    // Method to delete a student by ID
    public void deleteStudent(Long id) {
        studentRepository.deleteById(id);
    }
}

File: studentservice.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8080/api/students';

const getAllStudents = async () => {
    try {
        const response = await axios.get(BASE_URL);
        return response.data;
    } catch (error) {
        console.error('There was an error retrieving the students!', error);
        throw error;
    }
};

const getStudentById = async (id) => {
    try {
        const response = await axios.get(`${BASE_URL}/getById/${id}`);
        return response.data;
    } catch (error) {
        console.error(`There was an error retrieving the student with ID ${id}!`, error);
        throw error;
    }
};

const createStudent = async (student) => {
    try {
        const response = await axios.post(`${BASE_URL}/add`, student);
        return response.data;
    } catch (error) {
        console.error('There was an error creating the student!', error);
        throw error;
    }
};

const updateStudent = async (id, student) => {
    try {
        const response = await axios.put(`${BASE_URL}/update/${id}`, student);
        return response.data;
    } catch (error) {
        console.error(`There was an error updating the student with ID ${id}!`, error);
        throw error;
    }
};

const deleteStudent = async (id) => {
    try {
        await axios.delete(`${BASE_URL}/delete/${id}`);
    } catch (error) {
        console.error(`There was an error deleting the student with ID ${id}!`, error);
        throw error;
    }
};

export {
    getAllStudents,
    getStudentById,
    createStudent,
    updateStudent,
    deleteStudent
};
